pkg/image/registry/imagestream/strategy.go

```diff
@@ -156,12 +156,6 @@ func (s Strategy) tagsChanged(old, stream *api.ImageStream) fielderrors.Validati
 		}
 
 		if tagRef.From.Kind == "DockerImage" && len(tagRef.From.Name) > 0 {
-			event, err := tagReferenceToTagEvent(stream, tagRef, "")
-			if err != nil {
-				errs = append(errs, fielderrors.NewFieldInvalid(fmt.Sprintf("spec.tags[%s].from", tag), tagRef.From, err.Error()))
-				continue
-			}
-			api.AddTagEventToImageStream(stream, tag, *event)
```

Comment:
I think the description was confusing. With this PR, when you set spec.tags[2.2].from=openshift/ruby-22-centos7 (kind=DockerImage), you only get *1* image:

```yaml
  - items:
    - created: 2015-09-30T13:49:37Z
      dockerImageReference: openshift/ruby-20-centos7@sha256:509c5a47b689e1523388ff6c160b0ea1417f412794b4a9854277b272de7384d3
      image: sha256:509c5a47b689e1523388ff6c160b0ea1417f412794b4a9854277b272de7384d3
```